metricflow/specs/dimension_spec_resolver.py

```diff
@@ -50,5 +50,10 @@ def resolve_time_dimension_spec(
         return TimeDimensionSpec(
             element_name=call_parameter_set.time_dimension_reference.element_name,
             entity_links=call_parameter_set.entity_path,
-            time_granularity=call_parameter_set.time_granularity,
+            time_granularity=(
+                call_parameter_set.time_granularity
+                # TODO: This should be updated once resolution of unspecified grain is supported.
```

Comment:
Ok. DAY could be wrong, and then it'll throw a less helpful error, but at least it'll work for daily granularities and error out otherwise.